quipucords/api/source/model.py

```diff
@@ -69,8 +68,8 @@ class Source(models.Model):
     port = models.IntegerField(null=True)
     options = models.OneToOneField(SourceOptions, null=True, on_delete=models.CASCADE)
     credentials = models.ManyToManyField(Credential)
-    hosts = models.TextField(unique=False, null=False)
-    exclude_hosts = models.TextField(unique=False, null=True)
+    hosts = models.JSONField(unique=False, null=True)
```

Comment:
Why did this switch from `null=False` to `null=True`?

Reply:
Whenever any of the JSONField were null=False, was getting tons of errors like:
```
django.db.utils.IntegrityError: null value in column "metadata" of relation "api_systemfingerprint" violates not-null constraint
```

I think we're ok here since the serializers are helping us with the object existence and such via _required_=True when data is written.

Reply:
Hmm, interesting. 🤔 I feel like this would be worth revisiting this later, but for now, yeah we can probably trust the serializer to do its enforcement job. Hopefully there's no other way to write data without going through the serializer.

One last question before resolving this conversation: When did you encounter that error? Was it during the DB migration, running tests, and/or manually running the real service itself? My thought, which I don't think is right but I'll say it anyway, is that _maybe_ we need a two-step migration for these JSON fields. One to change the column/field type, and then one to set the not-null constraint. Did you try anything like that during development?

Reply:
Migrations to/from had no issues at all. The errors flood out while running the tests.

Not for the null constraint violations, but I've been playing with a field conversion (i.e. rename field to temporary name, create new/real new JSONField field name, convert/copy data over, then delete temporary name field) during the migration, I presume the same is needed for reverse migration.   I `think` I'll need this for the RawFact values unless JSONField performs some magic for us.

If you have some examples for the 2-step migrations you mentioned (change field, then set the not-null construct), I'll gladly give it a try.

Reply:
Oooh. I like having an empty list as a default.

Reply:
using an empty list might be a problem (as well using `{}`): you are saying the default value is a very specific list/dictionary already initialized in memory. 

Unless django does something to handle that issue, we would be mutating the same list/dict when default value is used.

Reply:
tldr: use default=list or default=dict

Reply:
👍

Reply:
Looks good! Thanks for the thorough feedback! 👍